gph/src/ripser.h

```diff
@@ -991,33 +991,35 @@ class ripser
         value_t birth, death;
         diameter_index_t birth_vertex;
         for (auto e : edges) {
-            get_simplex_vertices(get_index(e), 1, n, vertices_of_edge.rbegin());
-            index_t v1 = vertices_of_edge[0];
-            index_t v2 = vertices_of_edge[1];
-            index_t u = dset.find(v1), v = dset.find(v2);
-
-            if (u != v) {
-                birth_vertex = dset.link_roots_and_get_birth_vertex(u, v);
-                /* Unlike Ripser/ripser, here we do not exclude edges with
-                 * zero "diameters". This is because we don't assume that
-                 * all vertex weights are zero, as Ripser/ripser does. */
-                /* Elder rule; youngest class (max birth time of u and v)
-                 * dies first */
-                birth = birth_vertex.first;
-                death = get_diameter(e);
-                if (death > birth) {
-                    births_and_deaths_by_dim[0].push_back(birth);
-                    births_and_deaths_by_dim[0].push_back(death);
-
-                    if (return_flag_persistence_generators) {
-                        flag_persistence_generators.finite_0.push_back(
-                            {birth_vertex.second, std::max(v1, v2),
-                             std::min(v1, v2)});
+            if (get_diameter(e) != std::numeric_limits<value_t>::infinity()) {
```

Comment:
If an edge is infinity, it must be skipped (it means it does not exist).

Reply:
That is why we don't even need to do the link, right ?

Reply:
Let me suggest a slightly different code:

```cpp
if (std::isinf(get_diameter(e)))
    continue;
```

And then the rest comes back to the same level of indentation.

BTW, It was my fault, but I think we need to start to use `std::isinf` when comparing to infinity.

Reply:
I considered that option, but didn't implement it in the end because (as far as I understand), `isinf` will also yield `true` for negative infinity, but we don't want to `continue` in that case.  I know it would be very pathological input, but still...

Reply:
It is okay, I did not think about this case.
The yes, lets continue with `std::numeric_limits<value_t>::infinity()`.

Then, what do you think of updating the code like that :

```cpp
if (get_diameter(e) == std::numeric_limits<value_t>::infinity())
    continue;
```

Reply:
> That is why we don't even need to do the link, right ?

Right!

> Then, what do you think of updating the code like that :

I'm guessing it reduces indentation and divergence from Ripser/ripser, so why not!

Reply:
Done in f499f17. Maybe something similar can be done in other parts?